### PR TITLE
serviceData and manufacturerData filter members

### PR DESCRIPTION
**Pull request by zakorgy**
Updating `Bluetooth.webidl`. `serviceData` and `manufacturerData` informations can be passed when calling `requestDevice`.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

### CODE REVIEW

File: tests/wpt/mozilla/tests/mozilla/bluetooth/requestDevice/device-found-with-key-only.html [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<!doctype html>
+<meta charset="utf-8">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/_mozilla/mozilla/bluetooth/bluetooth-helpers.js"></script>
+<script>
+'use strict';
+window.testRunner.setBluetoothMockDataSet(adapter_type.glucose_heart_rate);
```
Comment (zakorgy) [inline tests/wpt/mozilla/tests/mozilla/bluetooth/requestDevice/device-found-with-key-only.html]: remove this line

File: components/script/dom/bluetooth.rs [hunk 2]
```diff
@@ -286,30 +293,67 @@ fn canonicalize_filter(filter: &BluetoothRequestDeviceFilter) -> Fallible<Blueto
...
+
+fn canonicalize_bluetooth_data_filter_init(bdfi: &BluetoothDataFilterInit) -> Fallible<(Vec<u8>, Vec<u8>)> {
+    let data_prefix = match bdfi.dataPrefix {
+        Some(ref d) => d.clone(),
+        None => Vec::new(),
+    };
+    let mask = match bdfi.mask {
+        Some(ref m) => m.clone(),
+        None => {
+            vec![0xff; data_prefix.len()]
```
Comment (dati91) [inline components/script/dom/bluetooth.rs]: remove the `{}`

File: tests/wpt/mozilla/tests/mozilla/bluetooth/requestDevice/canonicalizeFilter/wrong-mask-length.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+<meta charset="utf-8">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/_mozilla/mozilla/bluetooth/bluetooth-helpers.js"></script>
+<script>
+'use strict';
+let length_error_cases = [{
+    filters: [{
+        manufacturerData: {
+            17: {dataPrefix: new Uint8Array([0x91, 0xAA, 0x78]),
```
Comment (dati91) [inline tests/wpt/mozilla/tests/mozilla/bluetooth/requestDevice/canonicalizeFilter/wrong-mask-length.html]: nit: maybe you could improve the style with something like this

```
manufacturerData: {
    17: {
        dataPrefix: new Uint8Array([0x91, 0xAA, 0x78]),
        mask: new Uint8Array([0x0F, 0x57])
    }
}
```

File: tests/wpt/mozilla/tests/mozilla/bluetooth/requestDevice/device-found-with-key-only.html [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
```
Comment (dati91) [inline tests/wpt/mozilla/tests/mozilla/bluetooth/requestDevice/device-found-with-key-only.html]: remove the empty line

File: tests/wpt/mozilla/tests/mozilla/bluetooth/requestDevice/device-found-with-service-and-manufacturer-data.html [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<!doctype html>
+<meta charset="utf-8">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/_mozilla/mozilla/bluetooth/bluetooth-helpers.js"></script>
+<script>
+'use strict';
+
```
Comment (dati91) [inline tests/wpt/mozilla/tests/mozilla/bluetooth/requestDevice/device-found-with-service-and-manufacturer-data.html]: remove the empty line

File: components/script/dom/bluetooth.rs [hunk 6]
```diff
@@ -286,30 +293,67 @@ fn canonicalize_filter(filter: &BluetoothRequestDeviceFilter) -> Fallible<Blueto
...
+    let manufacturer_data = match filter.manufacturerData {
+        Some(ref manufacturer_data_map) => {
+            if manufacturer_data_map.deref().is_empty() {
+                return Err(Type(MANUFACTURER_DATA_ERROR.to_owned()));
+            }
+            let mut map = HashMap::new();
+            for (key, bdfi) in manufacturer_data_map.deref() {
+                let manufacturer_id = match u16::from_str(key.as_ref()) {
+                    Ok(id) => id,
+                    Err(err) => return Err(Type(format!("{:?} {:?} {:?}", KEY_CONVERSION_ERROR, key, err))),
```
Comment (dati91) [inline components/script/dom/bluetooth.rs]: You sure about `{:?}`, not just `{}`?

File: components/script/dom/bluetooth.rs (hunk 2 above)
Comment (dati91) [reply components/script/dom/bluetooth.rs]: and maybe add a comment about this

File: components/script/dom/webidls/Bluetooth.webidl [hunk 7]
```diff
@@ -4,16 +4,27 @@
...
+};
+
+dictionary BluetoothLEScanFilterInit {
   sequence<BluetoothServiceUUID> services;
   DOMString name;
   DOMString namePrefix;
-  unsigned short manufacturerId;
-  BluetoothServiceUUID serviceDataUUID;
+//  Maps unsigned shorts to BluetoothDataFilters.
+//  object manufacturerData;
```
Comment (dati91) [inline components/script/dom/webidls/Bluetooth.webidl]: i think you can remove this line and the other one as well

**Comment by dati91** [comment]
LGTM

**Pull request merged**